packages/config-loader/src/lib/env.ts

```diff
@@ -72,11 +72,16 @@ export function readEnv(env: {
             );
           }
           try {
-            const parsedValue = JSON.parse(value);
+            const [err, parsedValue] = safeJsonParse(value);
             if (parsedValue === null) {
               throw new Error('value may not be null');
             }
             obj[part] = parsedValue;
+            if (err) {
+              console.warn(
```

Comment:
Tbh I think we should skip the warning. I don't think this new behaviour is something for edge use-cases where you can't get quotes around the value, but rather the standard way that will be recommended through documentation.

We can prolly wait and see if it actually causes trouble for people to not have the warning. Tbh I think the cases for actually wanting a JSON-encoded non-primitive value are rare, and hopefully the failure is somewhat obvious if it happens.